### PR TITLE
improvements

### PR DESCRIPTION
**Pull request by treworld**
simplify configs, respect CDN domain, removed broken example, faster way to check existing images

### CODE REVIEW

**Comment by Jaza** [comment]
@treworld thanks for the improvements, overall they look good. However, re: "simplify configs", I'm afraid you're removing an important feature, which is the ability for each individual image to be stored with a different storage type / stored in a different bucket.

The idea with this, is that you can e.g. copy content from one environment to another (say, from "production" to "staging"), and the images will still "just work" (because the content copied from production is referencing images in the "production" S3 bucket). You could then add new images in "staging", and they'll get saved to the "staging" S3 bucket. So, one environment (e.g. "staging") could be managing images that are stored in multiple S3 buckets - each image knows which bucket it's stored in.

Hope that makes sense?

So, in a nutshell - can't accept this pull request, until you put back the `storage_type` and `bucket_name` parameters in `thumbnail()` and `_thumbnail_s3()`.
